sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -404,6 +386,35 @@ case class DataSource(
   }
 
   /**
+    * Creates Hadoop relation based on format and globbed file paths
+    * @param format format of the data source file
+    * @param globPaths Path to the file resolved by Hadoop library
+    * @return Hadoop relation object
+    */
+  def createHadoopRelation(format: FileFormat,
+                           globPaths: Array[Path]): BaseRelation = {
+    val (dataSchema, partitionSchema) = getOrInferFileFormatSchema(format)
```

Comment:
You do twice `getOrInferFileFormatSchema`. One is before calling `createHadoopRelation`.